### ISSUE TITLE
(v9) Badge: insufficient color contrast on high contrast mode

### ISSUE THREAD

**Issue by tringakrasniqi** [microsoft/fluentui#21127]
### Environment Information

- **Package version(s)**: v9 beta
- **Browser and OS versions**: Edge, Windows 11

### Describe the issue:

Using the light high contrast mode (desert) on Win 11:
- The icons on Badge do not meet the contrast ratio and are displayed as white: 
![white-HCM-icon](https://user-images.githubusercontent.com/12565826/147580829-618576c2-a6e0-424d-ba32-d5aa9919021e.png)

- The PresenceBadges on this theme are all shown with the same color, dark gray, but in dark high contrast mode the colors are shown properly (green, yellow etc.)
![WHCM_light_presencebadges](https://user-images.githubusercontent.com/12565826/147580899-59c97bf2-6eca-407b-b45f-8925ad54a08b.png)

### Please provide a reproduction of the issue in a codepen:
Codesandbox: https://codesandbox.io/s/badge-a11y-review-bhu70

**Comment by tringakrasniqi** [comment]
Cannot reproduce this bug, neither reverting the latest changes, this could have been because of storybook. I'll close the issue for now and if this comes up again I will investigate further.
